### PR TITLE
[WIP] Stadia GGP integration

### PR DESCRIPTION
**Pull request by beckerhe**
This pull requests adds additional functionality to the Qt UI.

Instead of providing the IP address to a Stadia instance via the command line,
there is now also a menu action (File -> Connect to Stadia Instance). This
action shows a dialog with all the currently assigned Stadia instances. The user
can pick one and connect to this instance.

This integration requires the GGP command line tool to be available and reachable,
so it needs to be in the path of the user's environment.

### CODE REVIEW

**Comment by ronaldfw** [comment]
I agree with Pierric that we should aim to make the UI more uniform. I think this should be embedded in the first tab of ORBIT (apologies, I should have mentioned this earlier). Then all functionality to select where and what to profile are on the first tab. The selection should also include local devices (e.g. the Windows workstation that one runs ORBIT on), in case this makes sense.

File: OrbitQt/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.8)
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply OrbitQt/CMakeLists.txt]: The cxx_std_11/cxx_std_14/cxx_std_17 compile features have been first defined in 3.8. Before 3.8 there was no semantically correct way to state the requirement of C++17.

**Comment by beckerhe** [comment]
> I agree with Pierric that we should aim to make the UI more uniform. I think this should be embedded in the first tab of ORBIT (apologies, I should have mentioned this earlier). Then all functionality to select where and what to profile are on the first tab. The selection should also include local devices (e.g. the Windows workstation that one runs ORBIT on), in case this makes sense.

Sounds like a good idea. But I'm not sure how to squeeze it into the first page. Do you have something in mind?

Another option would be to show a dialog before opening the main window.
There you choose your profiling target machine (either local or some Stadia instance) and afterwards
the main window opens. This would be in line with the current functionality in Orbit, since
reconnecting to a different instance does not work currently, but my changes somehow give that impression. And if I were integrating this on the first tab, this would also be a problem.

Of course, this dialog won't be shown, when you invoke Orbit with the gamelet command line option.

**Comment by ronaldfw** [comment]
> Another option would be to show a dialog before opening the main window.
> There you choose your profiling target machine (either local or some Stadia instance) and afterwards
> the main window opens. This would be in line with the current functionality in Orbit, since
> reconnecting to a different instance does not work currently, but my changes somehow give that impression. And if I were integrating this on the first tab, this would also be a problem.

Sounds good to me. @pierricgimmig What do you think?

**Comment by pierricgimmig** [comment]
> Sounds good to me. @pierricgimmig What do you think?

@ronaldfw, yeah that works!  Eventually we could add the option to connect to a non-ggp remote instance inside that dialog too.  We should probably keep the menu entry that makes the dialog show up on demand in case we want to profile a different instance without having to restart Orbit.

**Comment by beckerhe** [comment]
I'm putting this on hold for now, because there is much more what have to be designed before we can this properly. Just to name a few features which have to be aligned with this:

- Deploy Orbit onto the instance
- Start Orbit on the instance
- Handle SSH tunneling
- Start a package on the instance
- Prevent user from profiling non-game processes

**Comment by pierricgimmig** [comment]
Closing this for now until we find time to get back to it.

**Pull request closed without merging**